src/picologging/__init__.py

```diff
@@ -57,6 +62,32 @@ def __init__(self, fmt, defaults=None):
 }
 
 
+_lock = threading.RLock()
```

Comment:
Why is the Python Recursive lock implementation being added to the module?

It shouldn't be used by picologging. picologging uses STL recursive mutex in the handler API.

https://github.com/microsoft/picologging/blob/main/src/picologging/handler.cxx#L13

Reply:
Yes I've seen that and I'd tried to avoid this Lock but I guess it serves a different purpose.

This lock is for the shared access to module-level variables like `picologging._handlers` which are used by `fileConfig` and the `incremental` option in `dictcConfig`:
https://github.com/python/cpython/blob/53e6a9a7254bdcd0538580ba7d799cd453e2dca5/Lib/logging/config.py#L512

But now I see the real problem, the handler extension does not modify the module-level variable `_handlers` so it doesn't matter if we introduce the lock or `piclogging._handlers` as they will always be unchanged.

Should I just remove the lock and variables and add that `incremental` option is not supported in `dictConfig`?

Reply:
please mark `incremental` as not supported and raise a `ValueError` if it is set. This should thin down the code substantially.